pyquil/quilatom.py

```diff
@@ -500,7 +500,8 @@ def _expression_to_string(expression):
         if isinstance(expression.op2, BinaryExp) and not (
                 expression.precedence < expression.op2.precedence
                 or expression.precedence == expression.op2.precedence
-                and expression.associates in ('right', 'both')):
+                and expression.associates in ('right', 'both')) \
+                or ("pi" in right and right != "pi"):
```

Comment:
Could you explain to me why you need `"pi" in right` in this guard?

Reply:
In the case that `expression.op2` is a float, the above test will fail (not of type `BinaryExp`). But that is also precisely when `right` might evaluate to something like `pi/2`. If `right` contains a `pi` then we likely want to group it (unless it is just `pi`).

Reply:
Is it possible for `expression.op2` to be a `Parameter` here, like `%I_pitty_the_fool`? If so I guess you just get an extra set of harmless parens, but maybe an explict `float` check would be 🆒 .

Reply:
@appleby 🤣 

Alas, that will trigger the paren-ing. Fixing.